### PR TITLE
util: add Cubic pipeline & refactor Fixed pipeline

### PR DESCRIPTION
**Pull request by chavoosh**

### CODE REVIEW

**Comment by jefft0** [comment]
Hi Chavoosh. To test, I changed the Face in examples/browser/test-pipeline.html to new Face({host: "memoria.ndn.ucla.edu"}).

In the browser page, I set the Interest to /ndn/edu/ucla/remap/demo/ndn-js-test/named-data.net/project/ndn-ar2011.html/%FDT%F7n%9E
When I set the dropdown to Fixed Pipeline, it fetches the content.  But when I set the dropdown to Cubic Pipeline, the console says "Error 8: Reached the maximum number of retries (3) while retrieving segment #1".

This happens in both Firefox and Chrome.

**Comment by chavoosh** [comment]
> Hi Chavoosh. To test, I changed the Face in examples/browser/test-pipeline.html to new Face({host: "memoria.ndn.ucla.edu"}).
> 
> In the browser page, I set the Interest to /ndn/edu/ucla/remap/demo/ndn-js-test/named-data.net/project/ndn-ar2011.html/%FDT%F7n%9E
> When I set the dropdown to Fixed Pipeline, it fetches the content. But when I set the dropdown to Cubic Pipeline, the console says "Error 8: Reached the maximum number of retries (3) while retrieving segment #1".
> 
> This happens in both Firefox and Chrome.


Thanks for reporting this. The code was based on an assumption that the version number is not provided by user. Now this assumption is removed. I tested the reported scenario and it works.

File: js/util/pipeline-cubic.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ var RttEstimator = require('./rtt-estimator.js').RttEstimator;
 var Pipeline = require('./pipeline.js').Pipeline;
 var LOG = require('../log.js').Log.LOG;
 
+Log.LOG = 4;
```
Comment (jefft0) [inline js/util/pipeline-cubic.js]: I think this was put here for debugging. Can you remove it? (The library should not set the LOG level. That is for the application to set.)

Comment (chavoosh) [reply js/util/pipeline-cubic.js]: Oops... I forgot to remove it.

Comment (chavoosh) [reply js/util/pipeline-cubic.js]: I pushed the new code.

**Comment by jefft0** [comment]
OK. It passes my tests. Are you ready for me to merge it?

**Comment by chavoosh** [comment]
> OK. It passes my tests. Are you ready for me to merge it?

Sure.

**Pull request merged**